src/processor/handler.rs

```diff
@@ -193,21 +194,65 @@ pub trait TransactionContext {
         note = "please use `set_state_entry` or `set_state_entries` instead"
     )]
     /// set_state requests that each address in the provided map be
+    /// set in validator state to its corresponding value. set_state is deprecated, please use
+    /// set_state_entry to set_state_entries instead
+    ///
+    /// # Arguments
+    ///
+    /// * `entries` - entries are a hashmap where the key is an address and value is the data
+    fn set_state(&self, entries: HashMap<String, Vec<u8>>) -> Result<(), ContextError> {
+        self.set_state_entries(entries)
+    }
+
+    /// set_state_entry requests that the provided address is set in the validator state to its
+    /// corresponding value.
+    ///
+    /// # Arguments
+    ///
+    /// * `address` - address of where to store the data
+    /// * `data` - payload is the data to store at the address
+    fn set_state_entry(&self, address: String, data: Vec<u8>) -> Result<(), ContextError>;
+
+    /// set_state_entries requests that each address in the provided map be
     /// set in validator state to its corresponding value.
     ///
     /// # Arguments
     ///
-    /// * `entries` - entries are a hashmap where the key is and address and value is the data
-    fn set_state(&self, entries: HashMap<String, Vec<u8>>) -> Result<(), ContextError>;
+    /// * `entries` - entries are a hashmap where the key is an address and value is the data
+    fn set_state_entries(&self, entries: HashMap<String, Vec<u8>>) -> Result<(), ContextError>;
 
     /// delete_state requests that each of the provided addresses be unset
+    /// in validator state. A list of successfully deleted addresses is returned.
+    /// delete_state is deprecated, please use delete_state_entry to delete_state_entries instead
+    ///
+    /// # Arguments
+    ///
+    /// * `addresses` - the addresses to delete
+    #[deprecated(
+        since = "0.3.0",
+        note = "please use `delete_state_entry` or `delete_state_entries` instead"
+    )]
+    fn delete_state(&self, addresses: &[String]) -> Result<Vec<String>, ContextError> {
```

Comment:
This does not match the old signature, (which was very wrong).  The deprecated method perhaps could maintain the old version.

Reply:
We have already fixed get_state, I suggest also fixing delete_state.